src/main/java/org/folio/circulation/resources/ItemsByInstanceResource.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.circulation.resources;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.circulation.domain.SearchInstance;
+import org.folio.circulation.infrastructure.storage.SearchRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.http.server.JsonHttpResponse;
+import org.folio.circulation.support.http.server.WebContext;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+
+public class ItemsByInstanceResource extends Resource {
+
+  private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+  public ItemsByInstanceResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    router.get("/circulation/items-by-instance")
+      .handler(this::getInstanceItems);
+  }
+
+  private void getInstanceItems(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+    List<String> queryParams = routingContext.queryParam("query");
+    if (!queryParams.isEmpty()) {
+      final var searchRepository = new SearchRepository(clients);
+      searchRepository.getInstanceWithItems(queryParams.get(0))
+        .thenApply(r -> r.map(this::toJson))
+        .thenApply(r -> r.map(JsonHttpResponse::ok))
+        .thenAccept(context::writeResultToHttpResponse);
+    }
+  }
+
+  private JsonObject toJson(SearchInstance instanceExtended) {
+    log.debug("toJson:: instanceExtended: {}", instanceExtended);
+    if (instanceExtended != null)
```

Comment:
I don't think that your instance can be null at this stage. And please don't use if-statements without curly braces.